src/asset_manager/mod.rs

```diff
@@ -215,14 +234,55 @@ impl AssetLoader {
     /// On desktop, it just loads an asset from
     /// the "assets" folder, on android it will
     /// load it from the embedded assets.
-    pub fn load_default<T, F>(&self, name: &str, format: F) -> AssetFuture<T>
+    pub fn load_default<T, F>(&self, name: &str, format: F)
         where T: Asset + 'static,
               T::Data: Send + 'static,
               T::Error: Send + 'static,
-              F: AssetFormat + Import<T::Data> + Send + 'static
+              F: AssetFormat + Import<T::Data> + Clone + Send + 'static
     {
         let store = DefaultStore;
-        self.load(&store, name, format)
+        self.load::<T, _, F>(&store, name, format);
+    }
+
+    /// Returns the number of assets
+    /// which were loaded by this asset loader.
+    pub fn num_assets(&self) -> usize {
+        self.assets.len()
+    }
+
+    /// Process a number of asset datasets
+    /// as long as `as_long` returns true.
+    ///
+    /// Intended to be used from the internals
+    /// of the engine to allow loading assets
+    /// as long as the wished frame time wasn't reached
+    /// yet.
+    pub fn process<F: FnMut() -> bool>(&mut self, context: &mut Context, mut as_long: F) {
```

Comment:
I changed asset management quite a bit, so that it now supports the features requested by @nchashch. Additionally, I think this implementation is quite nice: If there is still time left in this frame, then assets will be created as long as we didn't reach `(1/FPS)*0.95`. If the computer is too slow for the requested frames per second, it will only load one asset in order not slow it down too much